### PR TITLE
Upgrade to tf_0.12. Fix api_gateway variable naming. Update docs. Testing

### PR DESCRIPTION
**Pull request by Komorebi-E**
## Description

- Upgrades the modules to use terraform 0.12 HCL.
- For the `api_gateway`, the `5XXError` variables have had a name change, as they can't start with a number.
- Removed duplicate alarm definitions so code would pass terraform validation.

- The `elasticsearch` module has been tested.
- N.B. The other modules need testing.

## Related Issue

https://github.com/skyscrapers/puppet/issues/4569

## Motivation and Context

Modules need upgrading along with customer code and other modules that are on 0.12

## How Has This Been Tested?

With Customer code that used release `v1.2.1` for ES monitoring.

```
module "es_monitoring" {
  source                     = "github.com/skyscrapers/terraform-cloudwatch//elasticsearch?ref=df6307578caecdb110e12381a8f61f90e7c94841"
  elasticsearch_cluster_name = module.elasticsearch.domain_name
  aws_clientid               = "ACCOUNT_ID"
  sns_topic_arn              = data.terraform_remote_state.CUSTOMER.outputs.sns_topic_arn
}
```

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,5 +1,7 @@
 # terraform-cloudwatch
 
+N.B. Releases >= 2.0.0 been upgraded to use Terraform 0.12.?
```
Comment (iuriaranda) [inline README.md]: Why the exclamation mark?

File: README.md [hunk 2]
```diff
@@ -9,7 +11,9 @@ Terraform module to setup cloudwatch alerts and push them to SNS. This repositor
 
 ## API-Gateway
 
-Creates general alerts for the API-Gateway
+Creates general alerts for the API-Gateway.
+
+N.B. Releases >= 2.0.0 the `5XXError...` variables are now `fiveXXError...`.
```
Comment (iuriaranda) [inline README.md]: Why the name change?

File: elasticsearch/variables.tf [hunk 3]
```diff
@@ -5,28 +5,29 @@ variable "sns_topic_arn" {
...
 variable "aws_clientid" {
   description = "AWS ID of the account where the elasticsearch domain is running"
-  type        = "string"
+  type        = string
 }
 
 variable "free_storage_space_threshold" {
   description = "The minimum amount of available storage space in MegaByte."
-  type        = "string"
+  type        = string
```
Comment (iuriaranda) [inline elasticsearch/variables.tf]: These should be `number` now

**Review by iuriaranda — CHANGES_REQUESTED** [review]
A part from the individual comments, I think all `..._period` and `..._threshold` variables should be type `number`

File: README.md (hunk 2 above)
Comment (Komorebi-E) [reply README.md]: > For the api_gateway, the 5XXError variables have had a name change, as they can't start with a number.

Comment (Komorebi-E) [reply README.md]: > For the api_gateway, the 5XXError variables have had a name change, as they can't start with a number.

File: README.md (hunk 1 above)
Comment (Komorebi-E) [reply README.md]: Changed to an `n` to make it less ambiguous.

**Review by simonrondelez — APPROVED** [review]

**Review by venturel — APPROVED** [review]

**Pull request merged**